### PR TITLE
Add support for pytest 8.1 + remove support for 7.1.

### PR DESCRIPTION
**Pull request by icemac**
This will get green when there will be a new 8.1 release as 8.1.0 got yanked.

### CODE REVIEW

**Comment by hugovk** [comment]
Please ping when a new pytest 8.1 is out and the CI is green.

**Comment by icemac** [comment]
@hugovk We now got a pytest 8.1.1 release, so this PR is ready to be reviewed.

File: CHANGES.rst [hunk 1]
```diff
@@ -18,11 +18,15 @@ Breaking changes
 
 - Drop support for pytest 7.0.
 
+- Drop support for pytest 7.1.
```
Comment (hugovk) [inline CHANGES.rst:21]:
```suggestion
- Drop support for pytest < 7.2.
```

File: CHANGES.rst [hunk 2]
```diff
@@ -18,11 +18,15 @@ Breaking changes
...
+
 Features
 ++++++++
 
 - Add support for pytest 8.0.
 
+- Add support for pytest 8.1.
```
Comment (hugovk) [inline CHANGES.rst:28]:
```suggestion
- Add support for pytest 8.0, 8.1.
```

**Review by hugovk — APPROVED** [review]

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Comment by hugovk** [comment]
Thank you for the PR :)

**Pull request merged**